annnufan/assignment2/solution_2_test.cpp

```diff
@@ -2,12 +2,11 @@
 #include <cassert>
 
 int main() {
-	node* root = new node(7, new node(3, new node(2, new node(1), new node(6)), new node(5)), new node(4, nullptr, new node(8)));
-	node* empty_root = nullptr;
-	assert(find_lca(7, 8, root) == 7);
-	assert(find_lca(10, 9, empty_root) == -1);
-	assert(find_lca(6, 9, root) == -1);
-	assert(find_lca(1, 6, root) == 2);
-	assert(find_lca(8, 6, root) == 7);
+	node root(7, new node(3, new node(2, new node(1), new node(6)), new node(5)), new node(4, nullptr, new node(8)));
+	assert(find_lowest_common_ancestor(7, 8, &root) == 7);
+	assert(find_lowest_common_ancestor(10, 9, nullptr) == -1);

```

Comment:
In my opinion the line:
`node* empty_root = nullptr;`
helped readability because this made this line clear.
(`find_lowest_common_ancestor(10,9,empty_tree);` make the intention of this test very clear).

But `nullptr` is also ok (the reader at this point should know that `nullptr` represents an empty tree)